cmd/zetae2e/local/bitcoin.go

```diff
@@ -5,71 +5,128 @@ import (
+	// withdraw tests DON'T use Bitcoin node wallet
+	account = conf.AdditionalAccounts.UserBitcoinWithdraw
+	runnerWithdraw := initRunner(
+		"btc_withdraw",
+		account,
+		conf,
+		deployerRunner,
+		color.FgHiYellow,
+		verbose,
+		initNetwork,
+		false,
+	)
+
+	// initialize funds
+	// send BTC to TSS for gas fees and to tester ZEVM address
+	if initNetwork {
+		// mine 101 blocks to ensure the BTC rewards are spendable
+		// Note: the block rewards can be sent to any address in here
+		_, err := runnerDeposit.GenerateToAddressIfLocalBitcoin(101, runnerDeposit.BTCDeployerAddress)
+		require.NoError(runnerDeposit, err)
 
-		// funding the account
+		// send BTC to ZEVM addresses
+		runnerDeposit.DepositBTC(runnerDeposit.EVMAddress())
+		runnerDeposit.DepositBTC(runnerWithdraw.EVMAddress())
+	}
+
+	// create test routines
+	routineDeposit := createTestRoutine(runnerDeposit, depositTests)
+	routineWithdraw := createTestRoutine(runnerWithdraw, withdrawTests)
+
+	return routineDeposit, routineWithdraw
+}
+
+// initRunner initializes the runner for given test name and account
+func initRunner(
+	name string,
+	account config.Account,
+	conf config.Config,
+	deployerRunner *runner.E2ERunner,
+	printColor color.Attribute,
+	verbose, initNetwork, createWallet bool,
+) *runner.E2ERunner {
+	// initialize runner for bitcoin test
+	runner, err := initTestRunner(name, conf, deployerRunner, account, runner.NewLogger(verbose, printColor, name))
+	testutil.NoError(err)
+
+	// setup TSS address and setup deployer wallet
+	runner.SetupBitcoinAccounts(createWallet)
+
+	// initialize funds
+	if initNetwork {
+		// send some BTC block rewards to the deployer address
+		_, err = runner.GenerateToAddressIfLocalBitcoin(4, runner.BTCDeployerAddress)
+		require.NoError(runner, err)
+
+		// send ERC20 token on EVM
 		txERC20Send := deployerRunner.SendERC20OnEvm(account.EVMAddress(), 1000)
-		bitcoinRunner.WaitForTxReceiptOnEvm(txERC20Send)
+		runner.WaitForTxReceiptOnEvm(txERC20Send)
 
-		// depositing the necessary tokens on ZetaChain
-		txEtherDeposit := bitcoinRunner.DepositEther()
-		txERC20Deposit := bitcoinRunner.DepositERC20()
+		// deposit ETH and ERC20 tokens on ZetaChain
+		txEtherDeposit := runner.DepositEther()
+		txERC20Deposit := runner.DepositERC20()
 
-		bitcoinRunner.WaitForMinedCCTX(txEtherDeposit)
-		bitcoinRunner.WaitForMinedCCTX(txERC20Deposit)
+		runner.WaitForMinedCCTX(txEtherDeposit)
+		runner.WaitForMinedCCTX(txERC20Deposit)
+	}
 
-		bitcoinRunner.SetupBitcoinAccount(initBitcoinNetwork)
-		bitcoinRunner.DepositBTC()
+	return runner
+}
 
-		// run bitcoin test
-		// Note: due to the extensive block generation in Bitcoin localnet, block header test is run first
-		// to make it faster to catch up with the latest block header
-		testsToRun, err := bitcoinRunner.GetE2ETestsToRunByName(
+// createTestRoutine creates a test routine for given test names
+func createTestRoutine(r *runner.E2ERunner, testNames []string) func() error {
```

Comment:
`createBitcoinTestRoutine`? since this has bitcoin specific logging?

Reply:
renamed https://github.com/zeta-chain/node/pull/3105/commits/197553296025514e99f64ed4a54978d635fc7df1